pkg/roachprod/ui/spinner.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package ui
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"golang.org/x/sync/errgroup"
+)
+
+type Spinner struct {
+	msg          syncutil.AtomicString
+	tickCallback func()
+	quiet, term  bool
+	out          io.Writer
+	waitGroup    sync.WaitGroup
+	stopChan     chan struct{}
+	started      bool
+	mu           syncutil.Mutex
+}
+
+type CountingSpinner struct {
+	*Spinner
+	baseMsg       syncutil.AtomicString
+	count         int32
+	expectedTotal int
+}
+
+type SpinnerGroup struct {
+	errgroup.Group
+	spinner   *CountingSpinner
+	completed int32
+}
+
+var spinnerLoop = []string{"|", "/", "-", "\\"}
```

Comment:
`var spinnerLoop = []string{"⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"}`

Do you think this would be too crazy? 😆  (I stole it from the gceworker spinner)

Reply:
I had the exact same thought! I am 100% for this!